src/components/oculus-touch-controls.js

```diff
@@ -0,0 +1,199 @@
+var registerComponent = require('../core/component').registerComponent;
+var bind = require('../utils/bind');
+
+// FIXME: need appropriate models, not the vive ones!
+var TOUCH_CONTROLLER_MODEL_OBJ_URL = 'https://cdn.aframe.io/controllers/vive/vr_controller_vive.obj';
+var TOUCH_CONTROLLER_MODEL_OBJ_MTL = 'https://cdn.aframe.io/controllers/vive/vr_controller_vive.mtl';
+
+/**
+ * Oculus Touch Controls Component
+ * Interfaces with Oculus Touch controllers and maps Gamepad events to
+ * common controller buttons: trackpad, trigger, grip, menu and system
+ * It loads a controller model and highlights the pressed buttons
+ */
+module.exports.Component = registerComponent('oculus-touch-controls', {
+  dependencies: ['tracked-controls'],
+
+  schema: {
+    idPrefix: { default: 'Oculus Touch' },
+    hand: {default: 'left'},
+    buttonColor: { default: '#FAFAFA' },  // Off-white.
+    buttonHighlightColor: {default: '#22D1EE'},  // Light blue.
+    model: {default: true},
+    rotationOffset: {default: -999} // use -999 as sentinel value to auto-determine based on hand
+  },
+
+  // buttonId
+  // 0 - trackpad
+  // 1 - trigger ( intensity value from 0.5 to 1 )
+  // 2 - grip
+  // 3 - menu ( dispatch but better for menu options )
+  // 4 - system ( never dispatched on this layer )
+  mapping: {
```

Comment:
https://twitter.com/BinaryLegend/status/784157621674663939

Again we get back to the nightmare that will occur for every higher level component having to know every controller in order to do mapping.  remember that `hand-controls` is not a canonical component that can be built upon like `tracked-controls` ...

My suggestion is this: define common events that vendor-specific controllers are expected to fire, and also fire namespaced vendor-specific events.  I would propose that `trigger`, `grip` and `menu` be considered common.  I hear you about `trackpad`.  Any proposal for how to compose the namespaced vendor-specific events?  e.g. `oculus-touch-___` and `vive-___` ?